dids/diddht/diddht.go

```diff
@@ -1,175 +1,252 @@
-		default:
+}
+
+// PrivateKey is used to add a private key to the DID being created with the [Create] function.
+// Each PrivateKey provided will be used to generate a private key in the key manager and then
+// added to the DID Document as a VerificationMethod.
+func PrivateKey(algorithmID string, purposes ...didcore.Purpose) CreateOption {
+	return func(o *createOptions) {
+		keyOpts := privateKeyOption{algorithmID: algorithmID, purposes: purposes}
+
+		if o.privateKeys == nil {
+			o.privateKeys = make([]privateKeyOption, 0)
 		}
+
+		o.privateKeys = append(o.privateKeys, keyOpts)
 	}
+}
 
-	return document, nil
+// KeyManager is used to set the key manager that will be used to generate the private keys for the DID.
+func KeyManager(km crypto.KeyManager) CreateOption {
+	return func(o *createOptions) {
+		o.keyManager = km
+	}
 }
 
-// parseDNSDID takes the bytes of the DNS representation of a DID and creates an internal representation
-// used to create a DID document
-func parseDNSDID(data []byte) (*Decoder, error) {
-	var p dnsmessage.Parser
-	if _, err := p.Start(data); err != nil {
-		return nil, err
+// AlsoKnownAs is used to set the 'alsoKnownAs' property of the DID Document.
+// more details here: https://www.w3.org/TR/did-core/#also-known-as
+func AlsoKnownAs(aka ...string) CreateOption {
+	return func(o *createOptions) {
+		o.alsoKnownAs = aka
 	}
+}
 
-	didRecord := Decoder{
-		records: make(map[string]string),
+// Controllers is used to set the 'controller' property of the DID Document.
+// more details here: https://www.w3.org/TR/did-core/#controller
+func Controllers(controllers ...string) CreateOption {
+	return func(o *createOptions) {
+		o.controllers = controllers
 	}
+}
 
-	// need to skip questions to move the index to the right place to read answers
-	if err := p.SkipAllQuestions(); err != nil {
-		return nil, err
+// Relay sets the relay to use for publishing the DID to the DHT.
+func Relay(relayURL string, client *http.Client) CreateOption {
+	return func(o *createOptions) {
+		o.relay = pkarr.NewClient(relayURL, client)
 	}
+}
 
-	for {
-		h, err := p.AnswerHeader()
-		if errors.Is(err, dnsmessage.ErrSectionDone) {
-			break
-		}
+// Create creates a new `did:dht` DID and publishes it to the DHT network via a Pkarr relay.
+//
+// If no relay is passed in the options, Create uses a default Pkarr relay.
+// Spec: https://did-dht.com/#create
+func Create(opts ...CreateOption) (*did.BearerDID, error) {
+	return CreateWithContext(context.Background(), opts...)
+}
+
+// CreateWithContext creates a new `did:dht` DID and publishes it to the DHT network via a Pkarr relay.
+func CreateWithContext(ctx context.Context, opts ...CreateOption) (*did.BearerDID, error) {
 
-		if h.Type != dnsmessage.TypeTXT {
-			continue
+	// 0. Set default options
+	o := createOptions{
+		relay:       getDefaultRelay(),
+		keyManager:  crypto.NewLocalKeyManager(),
+		privateKeys: []privateKeyOption{},
```

Comment:
@mistermoe does this look accurate? I'm starting with no private keys (which reminds me, I should fail if no keys).

I know there's supposed tot be at least 1 verification method. I'm wondering if I should add it ONLY of the `opts` don't include any `PrivateKey`. Would that be accurate?

Reply:
the identity key is technically the one key you can guarantee will be created even if no additional private keys are provided by the called. the identity key is also a verification method

Reply:
looking at the current code, looks like you need to add the identity key as a vm, lmk if I have this wrong

> the identity key is also a verification method